Storage/tests/System/IamConfigurationTest.php

```diff
@@ -112,7 +105,126 @@ public function testCreateBucketWithUniformBucketLevelAccessAndInsertObject()
         $object->reload();
     }
 
-    private function bucketConfig($enabled = true)
+    /**
+     * @dataProvider publicAccessSetting
+     */
+    public function testTogglePublicAccessPrevention($enableInitially)
+    {
+        $expected = function($enabled) {
+            return $enabled ? 'enforced' : 'unspecified';
+        };
+
+        $bucket = self::createBucket(
+            self::$client,
+            uniqid(self::TESTING_PREFIX),
+            $this->papConfig($enableInitially)
+        );
+
+        $this->assertEquals(
+            $expected($enableInitially),
+            $bucket->info()['iamConfiguration']['publicAccessPrevention']
+        );
+
+        $bucket->update($this->papConfig(!$enableInitially));
+
+        $this->assertEquals(
+            $expected(!$enableInitially),
+            $bucket->info()['iamConfiguration']['publicAccessPrevention']
+        );
+    }
+
+    public function publicAccessSetting()
+    {
+        return [
+            [true],
+            [false]
+        ];
+    }
+
+    /**
+     * @expectedException Google\Cloud\Core\Exception\FailedPreconditionException
+     */
+    public function testSetBucketAclToPublicAccessFailsWithPublicAccessPrevention()
+    {
+        $bucket = self::createBucket(self::$client, uniqid(self::TESTING_PREFIX), $this->papConfig());
+        $bucket->acl()->add('allUsers', 'READER');
+    }
+
+    /**
+     * @expectedException Google\Cloud\Core\Exception\FailedPreconditionException
+     */
+    public function testSetObjectAclToPublicAccessFailsWithPublicAccessPrevention()
```

Comment:
This test is currently failing pending upstream fix.

Reply:
@jdpedrie is this test still failing?